Server/routes/api/sprint.js

```diff
@@ -11,14 +11,52 @@ const router = express.Router();
 const {Sprint} = require('../../models/Sprint');
 const {Card} = require('../../models/Card');
 
-router.route('/sprint')
+// View sprint by id
+router.get("/sprint/:id", (req, res) => {
+    Sprint.findById(req.params.id)
+      .then(sprint => res.json(sprint))
+      .catch(err => res.status(404).json({ success: false }));
+  });
+
+//remove sprint by id
+router.delete("/sprint/:id", (req, res) => {
+    Sprint.findById(req.params.id)
+      .then(sprint => sprint.remove().then(() => res.json({ success: true })))
+      .catch(err => res.status(404).json({ success: false }));
+  });
+
+  //edit sprint by id
+  router.put("/sprint/:id", async function(req,res){
+    let params = {};
+    for(let prop in req.body){
+        if(req.body[prop] = "index"){
+            res.status(200).send("WARNING: the sprint\'s index cannot be updated.");
+        }
+        else if(req.body[prop]){
+            params[prop] = req.body[prop];
+        }
+    };
 
-// Display data about a sprint
-.get(async function (req, res) {
+    let updatedSprint = false;
+    updatedSprint = await Sprint.findOneAndUpdate({'_id': req.params.id}, params);
+    if(updatedSprint){
+        res.status(200).send(`The sprint was updated: ${updatedSprint._id}`);
+    } else {
+        res.status(500).send('The sprint was not updated');
+    }
+});
 
-    let sprintCards = await Card.find({'sprint': req.body.id});
-    res.status(200).json(sprintCards);
-})
+router.route('/sprint')
+
+//get all sprint cards
+.get((req, res) => {
+    Card.find()
+    Card.find({project: req.body.project, 
+            "stage.stageName": { $in: ["TODO","WIP","VERIFICATION","DONE"]},
+            "stage.endDate": null})
+      .sort({ "stage.stateName": 1 })
```

Comment:
ERROR: stage.stageName, not stage.stateName
```suggestion
      .sort({ "stage.stageName": 1 })
```